#!/usr/bin/env python

#
# Copyright (C) 2012 Nexenta Systems, Inc.
# All rights reserved.
#
# This library is free software; you can redistribute it and/or
# modify it under the terms of the GNU Lesser General Public
# License as published by the Free Software Foundation; either
# version 2.1 of the License, or any later version.
#
# This library is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
# Lesser General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public
# License along with this library; if not, write to the Free Software
# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301 USA
#
# Author: legkodymov
import sys
import syslog

try:
    from lsm.nstor import NexentaStor
    from lsm.pluginrunner import PluginRunner

    if __name__ == '__main__':
        PluginRunner(NexentaStor, sys.argv).run()
except Exception as e:
    #This should be quite rare, but when it does happen this is pretty
    #key in understanding what happened, especially when it happens when running
    #from the daemon.
    msg = "Exception: %s\n" % str(e)
    syslog.syslog(syslog.LOG_ERR, msg)
    sys.stderr.write(msg)
    sys.exit(1)